src/ir_York.cpp

```diff
@@ -0,0 +1,362 @@
+      break;
+    case kYorkAuto:
+      _.Fan = kYorkFanAuto;
+      break;
+    case kYorkCool:
+    case kYorkHeat:
+    default:
+      _.Fan = std::min(speed, kYorkFanAuto);
+  }
+}
+
+/// Get the current fan speed setting.
+/// @return The current fan speed.
+uint8_t IRYorkAc::getFan(void) const {
+  return _.Fan;
+}
+
+/// Convert a stdAc::fanspeed_t enum into it's native speed.
+/// @param[in] speed The enum to be converted.
+/// @return The native equivalent of the enum.
+uint8_t IRYorkAc::convertFan(const stdAc::fanspeed_t speed) {
+  switch (speed) {
+    case stdAc::fanspeed_t::kMin:
+    case stdAc::fanspeed_t::kLow:
+      return kYorkFanLow;
+    case stdAc::fanspeed_t::kMedium:
+      return kYorkFanMedium;
+    case stdAc::fanspeed_t::kHigh:
+    case stdAc::fanspeed_t::kMax:
+      return kYorkFanHigh;
+    default:
+      return kYorkFanAuto;
+  }
+}
+
+/// Convert a native fan speed into its stdAc equivalent.
+/// @param[in] speed The native setting to be converted.
+/// @return The stdAc equivalent of the native setting.
+stdAc::fanspeed_t IRYorkAc::toCommonFanSpeed(const uint8_t speed) {
+  switch (speed) {
+    case kYorkFanHigh:   return stdAc::fanspeed_t::kMax;
+    case kYorkFanMedium: return stdAc::fanspeed_t::kMedium;
+    case kYorkFanLow:    return stdAc::fanspeed_t::kMin;
+    default:             return stdAc::fanspeed_t::kAuto;
+  }
+}
+
+/// Set the temperature.
+/// @param[in] degrees The temperature in degrees celsius.
+void IRYorkAc::setTemp(const uint8_t degrees) {
+  uint8_t temp = std::max(kYorkMinTemp, degrees);
+  temp = std::min(kYorkMaxTemp, temp);
+  _.Temp = temp;
+}
+
+/// Get the current temperature setting.
+/// @return Get current setting for temp. in degrees celsius.
+uint8_t IRYorkAc::getTemp(void) const {
+  return _.Temp;
+}
+
+/// Set the On Timer value of the A/C.
+/// @param[in] nr_of_mins The number of minutes the timer should be.
+/// @note The timer time only has a resolution of 10 mins.
+/// @note Setting the On Timer active will cancel the Sleep timer/setting.
+void IRYorkAc::setOnTimer(const uint16_t nr_of_mins) {
+  _.OnTimer = nr_of_mins / 10;
+}
+
+/// Set the Off Timer value of the A/C.
+/// @param[in] nr_of_mins The number of minutes the timer should be.
+/// @note The timer time only has a resolution of 10 mins.
+/// @note Setting the Off Timer active will cancel the Sleep timer/setting.
+void IRYorkAc::setOffTimer(const uint16_t nr_of_mins) {
+  _.OffTimer = nr_of_mins / 10;
+}
+
+
+/// Get the On Timer setting of the A/C.
+/// @return The Nr. of minutes the On Timer is set for.
+uint16_t IRYorkAc::getOnTimer(void) const {
+  return _.OnTimer * 10;
+}
+
+/// Get the Off Timer setting of the A/C.
+/// @return The Nr. of minutes the Off Timer is set for.
+/// @note Sleep & Off Timer share the same timer.
+uint16_t IRYorkAc::getOffTimer(void) const {
+  return _.OffTimer * 10;
+}
+
+/// CRC16-16 (a.k.a. CRC-16-IBM)
+void IRYorkAc::calcChecksum() {
+int j;
```

Comment:
`j` is a terrible variable name. Use something useful/self-documenting. e.g.
```c++
uint16_t index;
```
etc.
In fact it can be delared later in the scope of the `for` loop later. See below.